test/blackbox-tests/test-cases/pkg/ocamlformat/ocamlformat-wrapper.t

```diff
@@ -7,8 +7,10 @@ Exercise running the ocamlformat wrapper command.
   $ make_ocamlformat_opam_pkg "0.26.2"
   $ make_project_with_dev_tool_lockdir
 
-  $ DUNE_CONFIG__LOCK_DEV_TOOL=enabled dune tools exec ocamlformat
```

Comment:
This environment variable turns on a hidden feature that automatically install a dev tool when you try to execute it.

We have been lamenting on the design of this being the default and have not reached a conclusion thusfar.

However in this case, I think we would still want `exec` to be able to install the dev tool when this feature is enabled.

Reply:
I think the behavior of installing dev tools automatically relates more to commands like `dune fmt`. For `dune tools` exec should always install the tool. There's not much point in telling the user do do something, if we know what needs to be done.

It's a bit like `dune exec` vs `dune build && dune exec`. You can do the build by hand if you want, for various reasons but `exec` will build anyway.